owasp-top10-2016-mobile/m5/panda_zap/server/config/viper.go

```diff
@@ -0,0 +1,24 @@
+package config
+
+import (
+	"github.com/spf13/viper"
+)
+
+// NewViper retuns a new Viper instance and an error
+func NewViper() (*viper.Viper, error) {
+
+	viperInstance := viper.New()
+	viperInstance.SetEnvPrefix("M5")
+	viperInstance.AutomaticEnv()
+
+	// API default values
+	viperInstance.SetDefault("port", 11005)
+
+	// Database default values
+	viperInstance.SetDefault("database_type", "local")
+
+	// Authentication default values
+	viperInstance.SetDefault("auth_type", "jwt")
```

Comment:
Cool, but do we really need those to be configurable? I'm not sure on how this would help the user in completing the exercise.

Anyways, isn't using Viper an overkill for this? :smile:

Reply:
I wanted it to be versatile. 

We never know if a user could come up with a cooler/better solution! 😁